### ISSUE TITLE
Vault Agent Auto-Auth File Sink to support formatting contents

### ISSUE THREAD

**Issue by mafonso** [hashicorp/vault#6395]
I often need to load the contents of the sinked file in a systemd unit. It requires some adicional commands to load it and an environment variable suitable for other hashicorp tools that read it from VAULT_TOKEN.  (envconsul, consul-template, nomad, vault)

It would be great to be able to format (or a simple template) the sink file contents. e.g 
`VAULT_TOKEN=%s` 

This would allow the systemd unit to source it via `EnvironmentFile=/path/to/sinked/file`

**Comment by jsmilani** [comment]
I second this request but my use case is a little different. My use case is to use consul-template with the 'vault_agent_token_file' config option but these do not work together when using 'wrap_ttl' option in the sink. The JSON output cannot be parsed by consul-template. I would like to be able to reformat the JSON output to strip it down to just the token.

Maybe 'wrap_ttl' wasn't really meant to be used with consul-template, but it seems desirable to wrap the token whenever writing to disk and even better when consul-template immediately consumes it.

**Comment by zmingxie** [comment]
Will this feature be added to the roadmap soon?

I just run into some issues with Nomad integration, and having this implemented will make my life a lot easier. Thanks!

**Comment by drawks** [comment]
Any word on this being added to the roadmap? I've run into both of the use cases outline by @jsmilani and @mafonso

**Comment by hsimon-hashicorp** [comment]
Hi folks! Is this still an issue in newer versions of Vault? Please let me know so I can bubble it up accordingly. Thanks!

**Comment by drawks** [comment]
> Hi folks! Is this still an issue in newer versions of Vault? Please let me know so I can bubble it up accordingly. Thanks!

Yes, there is no change in behavior which solves for either of the detailed use cases

**Comment by hsimon-hashicorp** [comment]
> > Hi folks! Is this still an issue in newer versions of Vault? Please let me know so I can bubble it up accordingly. Thanks!
> 
> Yes, there is no change in behavior which solves for either of the detailed use cases

Thank you for the quick reply! I'll check with product management on this. Appreciate it!
